core/object/object.cpp

```diff
@@ -1117,6 +1117,7 @@ Error Object::emit_signalp(const StringName &p_name, const Variant **p_args, int
 	Vector<const Variant *> bind_mem;
 
 	Error err = OK;
+	const Variant v_this = this;
```

Comment:
This can be a bit dangerous to do in some cases because an object might have died (as an example a reference)  and while emitting a signal it will re-reference.

I would just do it like this:

```C++
Variant appended_self;


// When you _actually_ need to add a pointer to it:

if (append_source) {
   if (unlikely(appended_self.get_type()==Variant::NIL)) {
     appended_self = this;
   }
    bind_mem.write[0] = &appended_self;
}
```

Reply:
Well actually as I mentioned, I would put the appended variant last, but you get the idea.